code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -109,6 +116,43 @@
 	. += emissive_appearance(icon, "locked", alpha = src.alpha)
 	. += locked ? "locked" : "unlocked"
 
+/obj/structure/closet/proc/animate_door(closing = FALSE)
+	if(door_anim_time)
+		if(!door_obj) door_obj = new
+		vis_contents |= door_obj
+		door_obj.icon = icon
+		door_obj.icon_state = "[icon_door || icon_state]_door"
+		is_animating_door = TRUE
+		var/num_steps = door_anim_time / world.tick_lag
+		for(var/step in 0 to num_steps)
+			var/angle = door_anim_angle * (closing ? 1 - (step/num_steps) : (step/num_steps))
+			var/matrix/door_transform = get_door_transform(angle)
+			var/door_state = angle >= 90 ? "[icon_door_override ? icon_door : icon_state]_back" : "[icon_door || icon_state]_door"
+			var/door_layer = angle >= 90 ? FLOAT_LAYER : ABOVE_MOB_LAYER
+
+			if(step == 0)
+				door_obj.transform = door_transform
+				door_obj.icon_state = door_state
+				door_obj.layer = door_layer
+			else if(step == 1)
+				animate(door_obj, transform = door_transform, icon_state = door_state, layer = door_layer, time = world.tick_lag, flags = ANIMATION_END_NOW)
+			else
+				animate(transform = door_transform, icon_state = door_state, layer = door_layer, time = world.tick_lag)
+		addtimer(CALLBACK(src,.proc/end_door_animation),door_anim_time,TIMER_UNIQUE|TIMER_OVERRIDE)
+
+/obj/structure/closet/proc/end_door_animation()
+	is_animating_door = FALSE
+	vis_contents -= door_obj
+	update_icon()
+	COMPILE_OVERLAYS(src)
```

Comment:
Is there a reason we don't let SSoverlays handle this?

Reply:
I'm not familiar! What would that look like here?

Reply:
assuming the call to `update_icon` causes a call to `src.add_overlay(...)`, nothing needs to be done. Just remove the `COMPILE_OVERLAYS(src)` call. You might want to check it works as intended after doing so.